### ISSUE TITLE
[webhook-service] v2.5.3, wont properly deploy a GKE cluster (handler pipeline-controller: Internal error occurred: failed calling webhook)

### ISSUE THREAD

**Issue by p4ranoidandro1d** [rancher/rancher#30442]
**What kind of request is this (question/bug/enhancement/feature request):**

Bug

**Steps to reproduce (least amount of steps as possible):**

Rancher v2.5.3 deployed on a GKE cluster, this one installed with google terraform provider.
I'm then using the rancher2 provider to deploy another GKE cluster (all the other cluster are meant to be deployed using rancher2 terraform provider, as the register feature only applies to EKS so far).

This configuration is used to deploy a new cluster:

    resource "rancher2_cluster" "rancher-cluster" {
      name           = "test-gke"
      description    = "gke cluster created by terraform and rancher2 provider"
      gke_config {
        cluster_ipv4_cidr                             = "10.99.16.0/21"
        credential                                    = base64decode(file("creds/serviceaccount.json"))
        disk_type                                     = "pd-standard"
        image_type                                    = "COS"
        ip_policy_cluster_ipv4_cidr_block             = "10.99.16.0/21"
        ip_policy_cluster_secondary_range_name        = "k8s-pod-99-16-0-21"
        ip_policy_node_ipv4_cidr_block                = "10.98.10.32/27"
        ip_policy_services_ipv4_cidr_block            = "10.99.24.0/21"
        ip_policy_services_secondary_range_name       = "k8s-svc-99-24-0-21"
        ip_policy_subnetwork_name                     = "k8s-nodes-subnet-98-10-32-27"
        locations                                     = ["europe-west1-a"]
        machine_type                                  = "n1-standard-1"
        maintenance_window                            = ""
        master_ipv4_cidr_block                        = "10.98.16.16/28"
        master_version                                = "1.16.15-gke.4300"
        network                                       = "https://www.googleapis.com/compute/v1/projects/test-net/global/networks/shared-vpc"
        node_pool                                     = ""
        node_version                                  = "1.16.15-gke.4300"
        oauth_scopes                                  = ["https://www.googleapis.com/auth/logging.write", "https://www.googleapis.com/auth/monitoring"]
        project_id                                    = "test-project"
        service_account                               = ""
        sub_network                                   = "https://www.googleapis.com/compute/v1/projects/test-net/regions/europe-west1/subnetworks/k8s-nodes-subnet-98-10-32-27"
        disk_size_gb                                  = 20
        enable_alpha_feature                          = false
        enable_auto_repair                            = true
        enable_auto_upgrade                           = false
        enable_horizontal_pod_autoscaling             = true
        enable_http_load_balancing                    = true
        enable_kubernetes_dashboard                   = false
        enable_network_policy_config                  = false
        enable_nodepool_autoscaling                   = true
        enable_private_endpoint                       = true
        enable_private_nodes                          = true
        enable_stackdriver_logging                    = true
        enable_stackdriver_monitoring                 = true
        ip_policy_create_subnetwork                   = false
        issue_client_certificate                      = false
        kubernetes_dashboard                          = false
        labels                                        = {}
        local_ssd_count                               = 0
        master_authorized_network_cidr_blocks         = ["10.98.0.0/15"]
        max_node_count                                = 4
        min_node_count                                = 2
        node_count                                    = 2
        preemptible                                   = true
        # region = europe-west1
        # resource_labels
        use_ip_aliases                                = true
        # taints
        zone                                          = "europe-west1-a"
      }
    }
    
**Result:**
Basically, the terraform will timeout.
In the meanwhile, after adding the master authorized network manually (not set by terraform) the cluster will stay in an Error status with the UI message :

**Cluster health check failed: deployments.apps "cattle-cluster-agent" not found**

In the rancher logs, we loop over these messages :

    gcp-rancher-5fb9cf68bf-cbx4k > rancher | 2020/12/11 16:01:54 [INFO] Creating globalRoleBindings for u-vwu2qn2dxq
    gcp-rancher-5fb9cf68bf-cbx4k > rancher | 2020/12/11 16:01:55 [ERROR] error syncing 'c-25jx9/p-zvxqq': handler pipeline-controller: Internal error occurred: failed calling webhook "rancherauth.cattle.io": Post https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation?timeout=10s: context deadline exceeded, handler system-image-upgrade-controller: upgrade cluster c-25jx9 system service logging failed: cluster c-25jx9 not ready, requeuing
    gcp-rancher-5fb9cf68bf-cbx4k > rancher | 2020/12/11 16:01:55 [ERROR] error syncing 'system-library': handler system-image-upgrade-catalog-controller: upgrade cluster c-25jx9 system service logging failed: cluster c-25jx9 not ready, requeuing
    gcp-rancher-5fb9cf68bf-cbx4k > rancher | 2020/12/11 16:01:55 [ERROR] error syncing 'install-uuid': handler copy-settings: the server could not find the requested resource, requeuing
    gcp-rancher-5fb9cf68bf-cbx4k > rancher | 2020/12/11 16:01:55 [ERROR] error syncing 'system-default-registry': handler copy-settings: the server could not find the requested resource, requeuing
    gcp-rancher-5fb9cf68bf-p8487 > rancher | 2020/12/11 16:01:59 [INFO] [mgmt-project-rbac-create] Creating creator clusterRoleTemplateBinding for user user-lk29f for cluster c-25jx9
    gcp-rancher-5fb9cf68bf-cbx4k > rancher | 2020/12/11 16:02:04 [ERROR] error syncing 'c-25jx9/p-zt5mk': handler pipeline-controller: Internal error occurred: failed calling webhook "rancherauth.cattle.io": Post https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation?timeout=10s: context deadline exceeded, handler system-image-upgrade-controller: upgrade cluster c-25jx9 system service logging failed: cluster c-25jx9 not ready, requeuing
    gcp-rancher-5fb9cf68bf-p8487 > rancher | 2020/12/11 16:02:14 [INFO] kontainerdriver googlekubernetesengine listening on address 127.0.0.1:35137
    gcp-rancher-5fb9cf68bf-p8487 > rancher | 2020/12/11 16:02:14 [INFO] kontainerdriver googlekubernetesengine stopped
    gcp-rancher-5fb9cf68bf-p8487 > rancher | 2020/12/11 16:02:14 [INFO] Creating globalRoleBindings for u-tgsaaqiqwy
    gcp-rancher-5fb9cf68bf-p8487 > rancher | 2020/12/11 16:02:24 [INFO] [mgmt-project-rbac-create] Creating creator clusterRoleTemplateBinding for user user-lk29f for cluster c-25jx9
    gcp-rancher-5fb9cf68bf-cbx4k > rancher | 2020/12/11 16:02:35 [INFO] Creating globalRoleBindings for u-vwu2qn2dxq
    gcp-rancher-5fb9cf68bf-cbx4k > rancher | 2020/12/11 16:02:36 [INFO] Creating globalRoleBindings for u-6iaxg23r2a
    gcp-rancher-5fb9cf68bf-cbx4k > rancher | 2020/12/11 16:02:36 [ERROR] error syncing 'system-library': handler system-image-upgrade-catalog-controller: upgrade cluster c-25jx9 system service logging failed: cluster c-25jx9 not ready, requeuing
    gcp-rancher-5fb9cf68bf-cbx4k > rancher | 2020/12/11 16:02:36 [ERROR] error syncing 'install-uuid': handler copy-settings: the server could not find the requested resource, requeuing
    gcp-rancher-5fb9cf68bf-cbx4k > rancher | 2020/12/11 16:02:36 [ERROR] error syncing 'system-default-registry': handler copy-settings: the server could not find the requested resource, requeuing
    gcp-rancher-5fb9cf68bf-p8487 > rancher | 2020/12/11 16:02:39 [INFO] kontainerdriver googlekubernetesengine listening on address 127.0.0.1:35547
    gcp-rancher-5fb9cf68bf-p8487 > rancher | 2020/12/11 16:02:39 [INFO] kontainerdriver googlekubernetesengine stopped
    gcp-rancher-5fb9cf68bf-p8487 > rancher | 2020/12/11 16:02:39 [INFO] Creating globalRoleBindings for u-tgsaaqiqwy

In the webhook logs, we see nothing. I tested the beta8, the same results.

**Other details that may be helpful:**

I'm quite new to Rancher/Kube, if there's information missing, i'll provide asap.

This is a GKE Shared VPC configuration, **private cluster**.
The connectivity is ok between the Rancher pod components and we can curl from the rancher/webhook pods to its services

**Environment information**
```
- Rancher version : v2.5.3
- Installation option (single install/HA): Rancher in GKE Kubernetes
 
```

**Cluster information**
```
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): 
    GKE 1.16 
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):
    $ kubectl version
    Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-11T13:17:17Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
    Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.15-gke.4300", GitCommit:"7ed5ddc0e67cb68296994f0b754cec45450d6a64", GitTreeState:"clean", BuildDate:"2020-10-28T09:23:22Z", GoVersion:"go1.13.15b4", Compiler:"gc", Platform:"linux/amd64"}

```

**Comment by p4ranoidandro1d** [comment]
Just FYI

- I got the same behavior with GKE 1.17
- Deploying Rancher v2.4.11 is ok with the first post scenario.

**Comment by p4ranoidandro1d** [comment]
#30251

**Comment by p4ranoidandro1d** [comment]
Resolved a few days ago, with the help of #30101 and rancher/terraform-provider-rancher2#548
